includes/abstracts/llms-abstract-user-engagement.php

```diff
@@ -232,10 +239,15 @@ public function sync( $context = 'update' ) {
 			delete_post_thumbnail( $this->get( 'post' ) );
 		}
 
+		// Replace references to reusable blocks with their actual blocks.
+		$content = $template->get( 'content', true );
+		$blocks  = parse_blocks( $content );
+		$blocks  = LLMS_Forms::instance()->load_reusable_blocks( $blocks );
+		$content = serialize_blocks( $blocks );
+
 		// Copy the content, with optional merge codes and short codes, and optional block editor layout meta properties
 		// from the template to this awarded engagement.
-		$this->set( 'content', $template->get( 'content', true ) );
-		$this->set( 'content', $this->merge_content() );
+		$this->set( 'content', $this->merge_content( $content ) );
```

Comment:
That's a good idea. I thought about making a stand-alone function with those 3 lines to avoid duplicate code and considered adding them to `merge_content()`, but at the time I was too lazy to deal with classes extended from `LLMS_Abstract_User_Engagement`. I see a way to do that easily now.